module-2/backend/week_two.md

```diff
@@ -5,37 +5,201 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+</br>
+</br>
+
 6. What's the difference between agile workflow and waterfall method?
+</br>
+
+ * Agile: Work happens in iterations. You plan for a small piece, build it out, test it, and then reconvene to plan the next piece and/or make changes to the previous iteration based on any new information we now have. 
+ * Waterfall: All planning, design, decisions, etc are made up front for a very specific product. Then the work is completed exactly to spec and is tested. There is no room for changing or updating the planned spec with waterfall methodology.
+</br>
+</br>
+
 7. What is the difference between `#new` and `#create`?
+</br>
+
+ * New instantiates a new object on a table, but does not save it unless told to do so.
+ * Create both instantiates and saves a new object on a table.
+</br>
+</br>
+
 8. At a basic level, what does cURL allow you to do?
+</br>
+cURL allows you to make GET and POST requests from the command line
+</br>
+</br>
+
 9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+</br>
+	Teachers
+	Has_many :students
+  
+  Students
+  Belongs_to :teachers
+
+
+</br>
+</br>
+
 10. Define foreign key, primary key, and schema.
+</br>
+
+* Primary key is the ID that is automatically assigned to a new item when it's created on a table.
+* The Foreign key is the value that is used on one table to identify a related item on a different table. For example, on a table of students, there might be a column for `teacher_id` that represents the teacher of that student where the `teacher_id` matches that teacher's primary key on the table of teachers.
+* The schema is created/updated when we do a migration. The schema creates our development database when we migrate and is also used as a blueprint for when we prepare our test database
+
+</br>
+</br>
+
 11. Describe the relationship between a foreign key on one table and a primary key on another table.
+</br>
+
+As noted above, the primary key is the identifier of an object on its own table where the foreign key is the identifier of an object that is being related to on a table other than its own.
+
+</br>
+</br>
+
 12. What are the parts of an HTTP response?
+</br>
+
+* Headers - This contains a lot of meta data with things like a timestamp, IP address, and format the body will be in.
+* Status Code - the code is a number between 100-600. Each number means something about whether the request was successful or not, and if not, gives a clue as to what went wrong.
+* Body - This is the information sent back; this may be empty or it may include content
+
+</br>
+</br>
+
 13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
+</br>
+
```

Comment:
This is on us. We didn't teach this time around, and should have deleted the question.